Framework/DataObjects/src/TimeSplitter.cpp

```diff
@@ -277,5 +290,122 @@ TimeROI TimeSplitter::getTimeROI(const int workspaceIndex) {
 
 std::size_t TimeSplitter::numRawValues() const { return m_roi_map.size(); }
 
+// ------------------------------------------------------------------------
+// SPLITTING EVENTS METHODS
+// ------------------------------------------------------------------------
+
+/**
+ * Split a list of events according to Pulse time or Pulse + TOF time
+ *
+ * Events with masked times are allocated to destination index -1.
+ * @param events : list of input events
+ * @param partials : resulting partial lists of events
+ * @param pulseTof : if True, split according to Pulse + TOF time, otherwise split by Pulse time
+ * @param tofCorrect : rescale and shift the TOF values (factor*TOF + shift)
+ * @param factor : rescale the TOF values by a dimensionless factor.
+ * @param shift : shift the TOF values after rescaling, in units of microseconds.
+ * @throws RunTimeError : the event list is of type Mantid::API::EventType::WEIGHTED_NOTIME
+ */
+void TimeSplitter::splitEventList(const EventList &events, std::map<int, EventList *> partials, bool pulseTof,
+                                  bool tofCorrect, double factor, double shift) const {
+
+  if (events.getEventType() == EventType::WEIGHTED_NOTIME)
+    throw std::runtime_error("EventList::splitByTime() called on an EventList "
```

Comment:
I think it would be clearer to make this [`std::invalid_argument`](https://en.cppreference.com/w/cpp/error/invalid_argument)